nixos/modules/programs/wayland/niri.nix

```
@@ -12,12 +12,21 @@ in
     enable = lib.mkEnableOption "Niri, a scrollable-tiling Wayland compositor";
 
     package = lib.mkPackageOption pkgs "niri" { };
+
+    xwaylandSatellite = {
+      enable = lib.mkEnableOption "Enable xwayland-satellite for Niri";
+
+      package = lib.mkPackageOption pkgs "xwayland-satellite" { };
+    };
   };
 
   config = lib.mkIf cfg.enable (
     lib.mkMerge [
       {
-        environment.systemPackages = [ cfg.package ];
+        environment.systemPackages = [
+          cfg.package
+        ]
+        ++ (lib.optional cfg.xwaylandSatellite cfg.xwaylandSatellite.package);
```

```suggestion
        ++ lib.optional cfg.xwaylandSatellite cfg.xwaylandSatellite.package;
```
unnecessary, function calls are prioritised before any other operator